executor/executor_pkg_test.go

```diff
@@ -83,10 +83,17 @@ func (s *testExecSuite) TestShowProcessList(c *C) {
 	ctx := context.Background()
 	// Run test and check results.
 	for _, p := range ps {
-		r, err := e.Next(ctx)
+		chk := e.newChunk()
+		err := e.NextChunk(context.Background(), chk)
 		c.Assert(err, IsNil)
-		c.Assert(r, NotNil)
-		c.Assert(r[0].GetUint64(), Equals, p.ID)
+		if chk.NumRows() != 0 {
+			var idx int
+			for rowIdx := 0; rowIdx < chk.NumRows(); rowIdx++ {
```

Comment:
We may use chunk.NewIterator4Chunk here.

Reply:
like this ?
```
+               for row := it.Begin(); row != it.End(); row = it.Next() {
                        var idx int
-                       for rowIdx := 0; rowIdx < chk.NumRows(); rowIdx++ {
-                               row := chk.GetRow(rowIdx)
+                       //for rowIdx := 0; rowIdx < chk.NumRows(); rowIdx++ {
+                       //      row := chk.GetRow(rowIdx)
                                c.Assert(row.GetInt64(int(p.ID)), Equals, int64(idx))
                                idx++
-                       }
+                       //}
                }

```

Reply:
yes

Reply:
review again, thanks @zz-jason @coocood @XuHuaiyu